cmd/serve/serve.go

```diff
@@ -243,23 +243,24 @@ func setupListeners(listenClientUrls []string) (listeners map[string]net.Listene
 
 // Serve starts the GRPC server on the provided socketPath. If the port is non-zero, it listens
 // to the TCP port instead of the unix domain socket.
-func Serve(log *logrus.Entry, listenClientUrls []string, s server.GRPCServer) error {
+func Serve(ctx context.Context, log *logrus.Entry, listenClientUrls []string, s server.GRPCServer) error {
 	clientListeners, err := setupListeners(listenClientUrls)
 	if err != nil {
 		return err
 	}
 
 	serverStart = time.Now()
-
 	if serveDuration != 0 {
 		// Register a server shutdown
-		go func() {
-			<-time.After(serveDuration)
-			log.WithField("duration", serveDuration).Info(
-				"Shutting down after the configured duration",
-			)
-			os.Exit(0)
-		}()
+		go func(ctx context.Context) {
+			select {
+			case <-time.After(serveDuration):
+				log.WithField("duration", serveDuration).Info(
+					"Shutting down after the configured duration",
+				)
```

Comment:
Just realized that the context should be cancelled here. Do we want to keep the server shutdown after duration feature though? What's the use-case for this?

Reply:
I believe the `serveDuration` was introduced to make integration tests simpler, but the integration tests never landed. However, given that we now have embedded Hubble in Cilium, this the `serveDuration` is becoming less and less useful, as even if we stop the Hubble server, `cilium-agent` will continue to run.

I agree with @michi-covalent (https://github.com/cilium/hubble/pull/182#pullrequestreview-371363840) that we should simply remove `serveDuration`.

Reply:
OK, I'll take care of this.